agg-ast/schema_derivation/src/schema_derivation.rs

```diff
@@ -2096,43 +2153,17 @@ impl DeriveSchema for UntaggedOperator {
                 }
                 Ok(Schema::simplify(&Schema::AnyOf(types)))
             }
-            // pow will return the maximal numeric type of its inputs; integrals are lumped together
-            // because an int ^ int can return a long
-            UntaggedOperatorName::Pow => {
-                let input_schema = get_input_schema(&args, state)?;
-                let mut types: BTreeSet<Schema> = set!();
-                if input_schema.satisfies(&Schema::Atomic(Atomic::Decimal)) != Satisfaction::Not {
-                    types.insert(Schema::Atomic(Atomic::Decimal));
-                }
-                if input_schema.satisfies(&Schema::Atomic(Atomic::Double)) != Satisfaction::Not{
-                    types.insert(Schema::Atomic(Atomic::Double));
-                }
-                if input_schema.satisfies(&Schema::Atomic(Atomic::Long)) != Satisfaction::Not {
-                    types.insert(Schema::Atomic(Atomic::Long));
-                }
-                if input_schema.satisfies(&Schema::Atomic(Atomic::Integer)) != Satisfaction::Not {
-                    types.insert(Schema::Atomic(Atomic::Integer));
-                    types.insert(Schema::Atomic(Atomic::Long));
-                };
-                handle_null_satisfaction(args, state, Schema::simplify(&Schema::AnyOf(types)))
-            }
             // mod returns the maximal numeric type of its inputs
```

Comment:
Here it states we're returning the maximal numeric and in the `$mod` docs that seems to be what we should do:
> Starting in version 7.2, the output data type of the $mod operator is the larger of the two input data types.
In the code I see it's combining all the numeric types, I could be missing something?

Reply:
You are right! I perhaps over simplified code here. I just pushed a change that incorporates a similar lower bound approach. We can't just re-use the same helper idt because we don't care about overflow here (ie if hte resulting type is int, there is no point making it int or long)